crates/ironrdp-displaycontrol/src/pdu/mod.rs

```diff
@@ -352,16 +374,17 @@ impl MonitorLayoutEntry {
     ///
     /// - `width` and `height` MUST be >= 200 and <= 8192.
     /// - `width` SHOULD be even. If it is odd, it will be adjusted
-    ///   to the nearest even number by subtracting 1.
+    ///    to the nearest even number by subtracting 1.
     fn new_impl(mut width: u32, height: u32) -> PduResult<Self> {
         if width % 2 != 0 {
             let prev_width = width;
             width = width.saturating_sub(1);
             warn!(
-                "Monitor width cannot be odd, adjusting from {} to {}",
+                "Monitor width cannot be odd, adjusting from [{}] to [{}]",
```

Comment:
I think I missed this part in the previous review, but now I am thinking that making silent adjustments and just printing warning in the low-level crate is not optimal. Maybe let's add a separate helper function `adjust_display_size(width, height) -> (u32, u32)`  in `ironrdp-displaycontrol` which could be called explicitly before monitor layout construction?

Reply:
bda1a96175175ea58f02e04fea89d12bc8a6d411